### ISSUE TITLE
Salesrule Validator argument error Magento 2.4.4  

### ISSUE THREAD

**Issue by cyildirim** [subscribepro/subscribepro-magento2-ext#145]
Following error occurs during add to cart request on Salesrule Validator Plugin. 
```
ArgumentCountError: Too few arguments to function Magento\SalesRule\Model\Validator::process(), 1 passed in /var/www/html/vendor/magento/framework/Interception/Interceptor.php on line 58 and exactly 2 expected in /var/www/html/vendor/magento/module-sales-rule/Model/Validator.php:295
Stack trace:
#0 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(58): Magento\SalesRule\Model\Validator->process(Object(Magento\Quote\Model\Quote\Item\Interceptor))
#1 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\SalesRule\Model\Validator\Interceptor->___callParent('process', Array)
#2 /var/www/html/vendor/subscribepro/subscribepro-magento2-ext/Plugin/SalesRule/Validator.php(67): Magento\SalesRule\Model\Validator\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Quote\Model\Quote\Item\Interceptor))
#3 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(135): Swarming\SubscribePro\Plugin\SalesRule\Validator->aroundProcess(Object(Magento\SalesRule\Model\Validator\Interceptor), Object(Closure), Object(Magento\Quote\Model\Quote\Item\Interceptor), Object(Magento\SalesRule\Model\Rule\Interceptor))
#4 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(153): Magento\SalesRule\Model\Validator\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Quote\Model\Quote\Item\Interceptor), Object(Magento\SalesRule\Model\Rule\Interceptor))
#5 /var/www/html/generated/code/Magento/SalesRule/Model/Validator/Interceptor.php(68): Magento\SalesRule\Model\Validator\Interceptor->___callPlugins('process', Array, NULL)
#6 /var/www/html/vendor/magento/module-sales-rule/Model/Quote/Discount.php(190): Magento\SalesRule\Model\Validator\Interceptor->process(Object(Magento\Quote\Model\Quote\Item\Interceptor), Object(Magento\SalesRule\Model\Rule\Interceptor))
#7 /var/www/html/vendor/magento/module-quote/Model/Quote/TotalsCollector.php(261): Magento\SalesRule\Model\Quote\Discount->collect(Object(Magento\Quote\Model\Quote\Interceptor), Object(Magento\Quote\Model\ShippingAssignment), Object(Magento\Quote\Model\Quote\Address\Total))
#8 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Quote\Model\Quote\TotalsCollector->collectAddressTotals(Object(Magento\Quote\Model\Quote\Interceptor), Object(Magento\Quote\Model\Quote\Address\Interceptor))
#9 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Quote\Model\Quote\TotalsCollector\Interceptor->___callParent('collectAddressT...', Array)
```

**Comment by garthbrantley** [comment]
More detailed stack trace:

```txt
rgumentCountError: Too few arguments to function Magento\SalesRule\Model\Validator::process(), 1 passed in /var/www/html/vendor/magento/framework/Interception/Interceptor.php on line 58 and exactly 2 expected in /var/www/html/vendor/magento/module-sales-rule/Model/Validator.php:295
Stack trace:
#0 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(58): Magento\SalesRule\Model\Validator->process(Object(Magento\Quote\Model\Quote\Item\Interceptor))
#1 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\SalesRule\Model\Validator\Interceptor->___callParent('process', Array)
#2 /var/www/html/app/code/Swarming/SubscribePro/Plugin/SalesRule/Validator.php(67): Magento\SalesRule\Model\Validator\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Quote\Model\Quote\Item\Interceptor))
#3 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(135): Swarming\SubscribePro\Plugin\SalesRule\Validator->aroundProcess(Object(Magento\SalesRule\Model\Validator\Interceptor), Object(Closure), Object(Magento\Quote\Model\Quote\Item\Interceptor), Object(Magento\SalesRule\Model\Rule))
#4 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(153): Magento\SalesRule\Model\Validator\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Quote\Model\Quote\Item\Interceptor), Object(Magento\SalesRule\Model\Rule))
#5 /var/www/html/generated/code/Magento/SalesRule/Model/Validator/Interceptor.php(68): Magento\SalesRule\Model\Validator\Interceptor->___callPlugins('process', Array, NULL)
#6 /var/www/html/vendor/magento/module-sales-rule/Model/Quote/Discount.php(190): Magento\SalesRule\Model\Validator\Interceptor->process(Object(Magento\Quote\Model\Quote\Item\Interceptor), Object(Magento\SalesRule\Model\Rule))
#7 /var/www/html/vendor/magento/module-quote/Model/Quote/TotalsCollector.php(261): Magento\SalesRule\Model\Quote\Discount->collect(Object(Magento\Quote\Model\Quote\Interceptor), Object(Magento\Quote\Model\ShippingAssignment), Object(Magento\Quote\Model\Quote\Address\Total))
#8 /var/www/html/generated/code/Magento/Quote/Model/Quote/TotalsCollector/Interceptor.php(41): Magento\Quote\Model\Quote\TotalsCollector->collectAddressTotals(Object(Magento\Quote\Model\Quote\Interceptor), Object(Magento\Quote\Model\Quote\Address\Interceptor))
#9 /var/www/html/vendor/magento/module-quote/Model/Quote/TotalsCollector.php(156): Magento\Quote\Model\Quote\TotalsCollector\Interceptor->collectAddressTotals(Object(Magento\Quote\Model\Quote\Interceptor), Object(Magento\Quote\Model\Quote\Address\Interceptor))
#10 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Quote\Model\Quote\TotalsCollector->collect(Object(Magento\Quote\Model\Quote\Interceptor))
#11 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Quote\Model\Quote\TotalsCollector\Interceptor->___callParent('collect', Array)
#12 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Quote\Model\Quote\TotalsCollector\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Quote\Model\Quote\Interceptor))
#13 /var/www/html/generated/code/Magento/Quote/Model/Quote/TotalsCollector/Interceptor.php(32): Magento\Quote\Model\Quote\TotalsCollector\Interceptor->___callPlugins('collect', Array, Array)
#14 /var/www/html/vendor/magento/module-quote/Model/Quote.php(2015): Magento\Quote\Model\Quote\TotalsCollector\Interceptor->collect(Object(Magento\Quote\Model\Quote\Interceptor))
#15 /var/www/html/generated/code/Magento/Quote/Model/Quote/Interceptor.php(743): Magento\Quote\Model\Quote->collectTotals()
#16 /var/www/html/vendor/magento/module-msrp/Block/Total.php(51): Magento\Quote\Model\Quote\Interceptor->collectTotals()
#17 /var/www/html/vendor/magento/framework/View/Element/AbstractBlock.php(1095): Magento\Msrp\Block\Total->_toHtml()
#18 /var/www/html/vendor/magento/framework/View/Element/AbstractBlock.php(1099): Magento\Framework\View\Element\AbstractBlock->Magento\Framework\View\Element\{closure}()
#19 /var/www/html/vendor/magento/framework/View/Element/AbstractBlock.php(660): Magento\Framework\View\Element\AbstractBlock->_loadCache()
#20 /var/www/html/vendor/magento/framework/View/Layout.php(578): Magento\Framework\View\Element\AbstractBlock->toHtml()
#21 /var/www/html/vendor/magento/framework/View/Layout.php(555): Magento\Framework\View\Layout->_renderBlock('checkout.cart.t...')
#22 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('checkout.cart.t...')
#23 /var/www/html/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('checkout.cart.t...')
#24 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('checkout.cart.t...', false)
#25 /var/www/html/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout\Interceptor->renderElement('checkout.cart.t...', false)
#26 /var/www/html/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('checkout.cart.t...', false)
#27 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('checkout.cart.t...')
#28 /var/www/html/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('checkout.cart.t...')
#29 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('checkout.cart.t...', false)
#30 /var/www/html/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout\Interceptor->renderElement('checkout.cart.t...', false)
#31 /var/www/html/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('cart.summary', false)
#32 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('cart.summary')
#33 /var/www/html/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('cart.summary')
#34 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('cart.summary', false)
#35 /var/www/html/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout\Interceptor->renderElement('cart.summary', false)
#36 /var/www/html/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('checkout.cart.c...', false)
#37 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('checkout.cart.c...')
#38 /var/www/html/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('checkout.cart.c...')
#39 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('checkout.cart.c...', false)
#40 /var/www/html/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout\Interceptor->renderElement('checkout.cart.c...', false)
#41 /var/www/html/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('checkout.cart.i...', false)
#42 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('checkout.cart.i...')
#43 /var/www/html/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('checkout.cart.i...')
#44 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('checkout.cart.i...', true)
#45 /var/www/html/vendor/magento/framework/View/Element/AbstractBlock.php(507): Magento\Framework\View\Layout\Interceptor->renderElement('checkout.cart.i...', true)
#46 /var/www/html/vendor/magento/module-checkout/view/frontend/templates/cart.phtml(16): Magento\Framework\View\Element\AbstractBlock->getChildHtml('with-items')
#47 /var/www/html/vendor/magento/framework/View/TemplateEngine/Php.php(71): include('/var/www/html/v...')
#48 /var/www/html/vendor/magento/framework/View/Element/Template.php(263): Magento\Framework\View\TemplateEngine\Php->render(Object(Magento\Checkout\Block\Cart), '/var/www/html/v...', Array)
#49 /var/www/html/vendor/magento/framework/View/Element/Template.php(293): Magento\Framework\View\Element\Template->fetchView('/var/www/html/v...')
#50 /var/www/html/vendor/magento/framework/View/Element/AbstractBlock.php(1095): Magento\Framework\View\Element\Template->_toHtml()
#51 /var/www/html/vendor/magento/framework/View/Element/AbstractBlock.php(1099): Magento\Framework\View\Element\AbstractBlock->Magento\Framework\View\Element\{closure}()
#52 /var/www/html/vendor/magento/framework/View/Element/AbstractBlock.php(660): Magento\Framework\View\Element\AbstractBlock->_loadCache()
#53 /var/www/html/vendor/magento/framework/View/Layout.php(578): Magento\Framework\View\Element\AbstractBlock->toHtml()
#54 /var/www/html/vendor/magento/framework/View/Layout.php(555): Magento\Framework\View\Layout->_renderBlock('checkout.cart')
#55 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('checkout.cart')
#56 /var/www/html/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('checkout.cart')
#57 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('checkout.cart', false)
#58 /var/www/html/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout\Interceptor->renderElement('checkout.cart', false)
#59 /var/www/html/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('content', false)
#60 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('content')
#61 /var/www/html/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('content')
#62 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('content', false)
#63 /var/www/html/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout\Interceptor->renderElement('content', false)
#64 /var/www/html/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('main', false)
#65 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('main')
#66 /var/www/html/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('main')
#67 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('main', false)
#68 /var/www/html/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout\Interceptor->renderElement('main', false)
#69 /var/www/html/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('columns', false)
#70 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('columns')
#71 /var/www/html/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('columns')
#72 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('columns', false)
#73 /var/www/html/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout\Interceptor->renderElement('columns', false)
#74 /var/www/html/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('main.content', false)
#75 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('main.content')
#76 /var/www/html/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('main.content')
#77 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('main.content', false)
#78 /var/www/html/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout\Interceptor->renderElement('main.content', false)
#79 /var/www/html/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('page.wrapper', false)
#80 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('page.wrapper')
#81 /var/www/html/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('page.wrapper')
#82 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('page.wrapper', false)
#83 /var/www/html/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout\Interceptor->renderElement('page.wrapper', false)
#84 /var/www/html/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('root', false)
#85 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('root')
#86 /var/www/html/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('root')
#87 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('root', true)
#88 /var/www/html/vendor/magento/framework/View/Layout.php(975): Magento\Framework\View\Layout\Interceptor->renderElement('root')
#89 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\View\Layout->getOutput()
#90 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\View\Layout\Interceptor->___callParent('getOutput', Array)
#91 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\View\Layout\Interceptor->Magento\Framework\Interception\{closure}()
#92 /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php(347): Magento\Framework\View\Layout\Interceptor->___callPlugins('getOutput', Array, Array)
#93 /var/www/html/vendor/magento/framework/View/Result/Page.php(260): Magento\Framework\View\Layout\Interceptor->getOutput()
#94 /var/www/html/vendor/magento/framework/View/Result/Layout.php(171): Magento\Framework\View\Result\Page->render(Object(Magento\Framework\App\Response\Http\Interceptor))
#95 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\View\Result\Layout->renderResult(Object(Magento\Framework\App\Response\Http\Interceptor))
#96 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\View\Result\Page\Interceptor->___callParent('renderResult', Array)
#97 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\View\Result\Page\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Response\Http\Interceptor))
#98 /var/www/html/generated/code/Magento/Framework/View/Result/Page/Interceptor.php(95): Magento\Framework\View\Result\Page\Interceptor->___callPlugins('renderResult', Array, Array)
#99 /var/www/html/vendor/magento/framework/App/Http.php(120): Magento\Framework\View\Result\Page\Interceptor->renderResult(Object(Magento\Framework\App\Response\Http\Interceptor))
#100 /var/www/html/generated/code/Magento/Framework/App/Http/Interceptor.php(23): Magento\Framework\App\Http->launch()
#101 /var/www/html/vendor/magento/framework/App/Bootstrap.php(264): Magento\Framework\App\Http\Interceptor->launch()
#102 /var/www/html/pub/index.php(30): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http\Interceptor))
#103 {main}
```

**Comment by garthbrantley** [comment]
Fixed in PR #153.
